R/util-func-namespace.R

```diff
@@ -0,0 +1,15 @@
+##' getFunctionIfNamespaced looks to see if a strFunction is namespaced and looks it up
+##' allowing the do.call in run step to process correctly
+##' this will either return a function, or the original string if not namespaced
+getStrFunctionIfNamespaced <- function(strFunction) {
+  fn_pieces <- strsplit(strFunction, "::")[[1]]
+  if (length(fn_pieces) > 2) {
+    rlang::abort("function pieces should only be namespaced via <pkg>::<fn>, detected more :: calls")
+  }
+  if (length(fn_pieces) == 2) {
+    pkg <- fn_pieces[[1]]
+    fn <- fn_pieces[[2]]
+    return(rlang::as_function(fn, env = getNamespace(pkg)))
+  }
+  return(strFunction) 
```

Comment:
```suggestion
  return(rlang::as_function(strFunction))
```
Oh yeah, I forgot you can use functions as strings in almost all situations (see `"mean"(1:10)`). Still, it probably wouldn't hurt to `rlang::as_function` this, to make sure it works as a function everywhere.

Reply:
i'd be completely fine with that, the reason i didn't do it was for expediency and I didn't know if there could be any reason that rlang::as_function could cause a behavioral shift compared to passing do.call a string. I waffled on adding that and figured I'd go conservative for time. If your experience is this also works the same then i'm all for it